gtsfm/frontend/detector_descriptor/detector_descriptor_base.py

```diff
@@ -43,9 +43,26 @@ def detect_and_describe(self, image: Image) -> Tuple[Keypoints, np.ndarray]:
             Corr. descriptors, of shape (N, D) where D is the dimension of each descriptor.
         """
 
+    def filter_by_response(self, keypoints: Keypoints, descriptors: np.ndarray) -> Tuple[Keypoints, np.ndarray]:
```

Comment:
why not just use `Keypoints.get_top_k()`?

https://github.com/borglab/gtsfm/blob/master/gtsfm/common/keypoints.py#L89

Reply:
Because we need the indices as well for the descriptors.